recipes/hdf4/build.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+if [[ $(uname) == Darwin ]]; then
+  export DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib
```

Comment:
@jakirkham I had to add this to succeed on OS X. (exporting at `make check` did not suffice BTW.)